lib/ttnt/test_selector.rb

```diff
@@ -11,11 +11,12 @@ class TestSelector
 
     # @param repo [Rugged::Reposiotry] repository of the project
     # @param target_sha [String] sha of the target object
+    #   (nil means to target current working tree)
     # @param test_files [#include?] candidate test files
     def initialize(repo, target_sha, test_files)
       @repo = repo
       @metadata = MetaData.new(repo, target_sha)
-      @target_obj = @repo.lookup(target_sha)
+      @target_obj = target_sha ? @repo.lookup(target_sha) : nil
```

Comment:
You can simply do:

~~~ruby
@target_obj = @repo.lookup(target_sha) if target_sha
~~~

instance variables are set to `nil` by default.